ospf6d/ospf6_bfd.c

```diff
@@ -254,70 +126,86 @@ static int ospf6_bfd_interface_dest_update(ZAPI_CALLBACK_ARGS)
 void ospf6_bfd_info_nbr_create(struct ospf6_interface *oi,
 			       struct ospf6_neighbor *on)
 {
-	struct bfd_info *oi_bfd_info;
-	struct bfd_info *on_bfd_info;
-
-	if (!oi->bfd_info)
+	if (!oi->bfd_config.enabled)
 		return;
 
-	oi_bfd_info = (struct bfd_info *)oi->bfd_info;
-
-	if (!on->bfd_info)
-		on->bfd_info = bfd_info_create();
-
-	on_bfd_info = (struct bfd_info *)on->bfd_info;
-	on_bfd_info->detect_mult = oi_bfd_info->detect_mult;
-	on_bfd_info->desired_min_tx = oi_bfd_info->desired_min_tx;
-	on_bfd_info->required_min_rx = oi_bfd_info->required_min_rx;
+	if (on->bfd_session == NULL)
+		on->bfd_session = bfd_sess_new(ospf6_bfd_callback, on);
+
+	bfd_sess_set_timers(on->bfd_session,
+			    oi->bfd_config.detection_multiplier,
+			    oi->bfd_config.min_rx, oi->bfd_config.min_tx);
+	bfd_sess_set_ipv6_addrs(on->bfd_session, on->ospf6_if->linklocal_addr,
+				&on->linklocal_addr);
+	bfd_sess_set_interface(on->bfd_session, oi->interface->name);
+	bfd_sess_set_profile(on->bfd_session, oi->bfd_config.profile);
+	bfd_sess_enable(on->bfd_session, false);
 }
 
 /*
  * ospf6_bfd_write_config - Write the interface BFD configuration.
  */
 void ospf6_bfd_write_config(struct vty *vty, struct ospf6_interface *oi)
 {
-#if HAVE_BFDD == 0
-	struct bfd_info *bfd_info;
-#endif /* ! HAVE_BFDD */
-
-	if (!oi->bfd_info)
+	if (!oi->bfd_config.enabled)
 		return;
 
 #if HAVE_BFDD == 0
-	bfd_info = (struct bfd_info *)oi->bfd_info;
-
-	if (CHECK_FLAG(bfd_info->flags, BFD_FLAG_PARAM_CFG))
+	if (oi->bfd_config.detection_multiplier != BFD_DEF_DETECT_MULT
+	    || oi->bfd_config.min_rx != BFD_DEF_MIN_RX
+	    || oi->bfd_config.min_tx != BFD_DEF_MIN_TX)
 		vty_out(vty, " ipv6 ospf6 bfd %d %d %d\n",
-			bfd_info->detect_mult, bfd_info->required_min_rx,
-			bfd_info->desired_min_tx);
+			oi->bfd_config.detection_multiplier,
+			oi->bfd_config.min_rx, oi->bfd_config.min_tx);
 	else
 #endif /* ! HAVE_BFDD */
-		vty_out(vty, " ipv6 ospf6 bfd\n");
+	{
+		if (oi->bfd_config.profile)
```

Comment:
Profile config should be displayed regardless of the other config.

Reply:
You mean the BFD should be displayed twice? One to enable BFD and a second time just for the profile?

Reply:
Yes. The config is displayed this way in BGP/OSPF, I think it should be consistent throughout the daemons.

Reply:
Fixed